toolchain/glibc.py

```diff
@@ -35,10 +35,17 @@ def setUp(self):
         for package in deps:
             if not sm.check_installed(package) and not sm.install(package):
                 self.error(package + ' is needed for the test to be run')
-        url = 'https://github.com/bminor/glibc/archive/master.zip'
-        tarball = self.fetch_asset("glibc.zip", locations=[url], expire='7d')
-        archive.extract(tarball, self.srcdir)
-        glibc_dir = os.path.join(self.srcdir, "glibc-master")
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            url = 'https://github.com/bminor/glibc/archive/master.zip'
+            tarball = self.fetch_asset("glibc.zip", locations=[url], expire='7d')
+            archive.extract(tarball, self.srcdir)
+            glibc_dir = os.path.join(self.srcdir, "glibc-master")
+        elif run_type == "distro":
+            glibc_dir = os.path.join(self.srcdir, "glibc-distro")
+            if os.path.exists(glibc_dir):
+                os.makedirs(glibc_dir)
```

Comment:
Why creating dir when path exists ?

Reply:
i think  condition is like may be typo
`if not os.path.exists(glibc_dir):
   os.makedirs(glibc_dir)`

Reply:
yeah, agreed.